File: easy/Remove_Duplicates_from_Sorted_Array.py
class Solution:
    def removeDuplicates(self, nums ) : 
        self.probe = None 
        self.k = 0
                
        for index in range(len(nums)): 
            if not self.probe == nums[index - self.k ]: 
                self.probe = nums[index - self.k ]
            else: 
                nums.pop(index - self.k ) 
                nums.append("_")
                self.k += 1 
                
        return len(nums) - self.k 

S = Solution()
number = [0,0,1,1,1,2,2,3,3,4]
print(S.removeDuplicates(nums=number))
print(number)
        

File: RUSH150/Remove Duplicates from Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    
    let insert_probe = 1 ;
    
    for (let check_probe = 1  ; check_probe<nums.length ; check_probe++){

        if (nums[check_probe] !== nums[check_probe-1] ){
            nums[insert_probe] = nums[check_probe] 
            insert_probe += 1 
        }
    }
    return insert_probe ;
};